src/_P022_PCA9685.ino

```diff
@@ -52,38 +67,203 @@ boolean Plugin_022(byte function, struct EventStruct *event, String& string)
-          SendStatus(event->Source, getPinStateJSON(SEARCH_PIN_STATE, PLUGIN_ID_022, event->Par1, log, 0));
+          log = String(F("PCA 0x")) + String(PCA9685_ADDRESS + port, HEX) + String(F(": GPIO ")) + String(event->Par1);
+          if(event->Par1 >= 0 && event->Par1 <= PCA9685_MAX_PINS)
+          {
+            if(event->Par2 >=0 && event->Par2 <= PCA9685_MAX_PWM)
+            {
+              if (!IS_INIT(initializeState, port)) Plugin_022_initialize(port);
+              
+              Plugin_022_Write(port, event->Par1, event->Par2);
+              setPinState(PLUGIN_ID_022, event->Par1, PIN_MODE_PWM, event->Par2);
+              addLog(LOG_LEVEL_INFO, log);
+              SendStatus(event->Source, getPinStateJSON(SEARCH_PIN_STATE, PLUGIN_ID_022, event->Par1, log, 0));
+            }
+            else{
+              addLog(LOG_LEVEL_ERROR, log + String(F(" the pwm value ")) + String(event->Par2) + String(F(" is invalid value.")));
+            }
+          }
+          else{
+            addLog(LOG_LEVEL_ERROR, log + String(F(" is invalid value.")));
+          }
         }
-        if (command == F("pcafrq"))
+        if (command == F("pcafrq") || (istanceCommand && command == F("frq")))
         {
-          if (!Plugin_022_init) Plugin_022_initialize();
           success = true;
-          Plugin_022_Frequency(event->Par1);
-          setPinState(PLUGIN_ID_022, 99, PIN_MODE_UNDEFINED, event->Par1);
-          log = String(F("PCA  : FREQ ")) + String(event->Par1);
-          addLog(LOG_LEVEL_INFO, log);
-          SendStatus(event->Source, getPinStateJSON(SEARCH_PIN_STATE, PLUGIN_ID_022, 99, log, 0));
+          if(event->Par1 >= PCA9685_MIN_FREQUENCY && event->Par1 <= PCA9685_MAX_FREQUENCY)
+          {          
+            if (!IS_INIT(initializeState, port)) Plugin_022_initialize(port);
+
+            Plugin_022_Frequency(port, event->Par1);
+            setPinState(PLUGIN_ID_022, 99, PIN_MODE_UNDEFINED, event->Par1);
+            log = String(F("PCA 0x")) + String(PCA9685_ADDRESS + port) + String(F(": FREQ ")) + String(event->Par1);
+            addLog(LOG_LEVEL_INFO, log);
+            SendStatus(event->Source, getPinStateJSON(SEARCH_PIN_STATE, PLUGIN_ID_022, 99, log, 0));
+          }
+          else{
+            addLog(LOG_LEVEL_ERROR,String(F("PCA ")) + String(PCA9685_ADDRESS + port, HEX) + String(F(" The frequesncy ")) + String(event->Par1) + String(F(" is out of range.")));
+          } 
+
         }
 
         if (command == F("status"))
         {
-          if (parseString(string, 2) == F("pca"))
+          if (parseString(line, 2) == F("pca"))
           {
-            if (!Plugin_022_init) Plugin_022_initialize();
+            if (!IS_INIT(initializeState, port)) Plugin_022_initialize(port);
             success = true;
             SendStatus(event->Source, getPinStateJSON(SEARCH_PIN_STATE, PLUGIN_ID_022, event->Par2, dummyString, 0));
           }
         }
+
+        if(dotPos >- 1 && command == F("gpio"))
```

Comment:
This line causes this issue: https://github.com/letscontrolit/ESPEasy/issues/1471
The name of the command has to be changed.